### PR TITLE
Add `todo` live template

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR creates `todo` template in `User` template group. 

It inserts a TODO comment with the current date and username. 

For example: `// TODO:2022-11-02:yevhenii.nadtochii: <...>`.

Unfortunately, IDEA [doesn't support](https://youtrack.jetbrains.com/issue/IDEA-242159/Would-nice-to-have-project-specific-live-templates) defining templates per-project basis (this feature request is 16 years old!). And it can be added only manually by following instructions I've left in the README file.

Along with that, this changeset does the following:

 1. Moves `Spine` template group to `config` to have all templates in a single place.
 2. Excludes `TODO` comments from Detekt reporting.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov Please take a look. Is this approach viable?

**Review by armiol — COMMENTED** [review]
@yevhenii-nadtochii 
Please see this: https://github.com/SpineEventEngine/documentation/wiki/IntelliJ-IDEA-configuration#live-templates

**Comment by armiol** [comment]
@yevhenii-nadtochii and this: https://sites.google.com/a/teamdev.com/engineers/development-standards/java/style-guide/todo-comments

File: .idea/live-templates/todo.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+-->
+
+<template name="todo" value="$LINE_COMMENT_START$ TODO:$DATE$:$USER$: $END$" description="Insert TODO comment with the current date and username" toReformat="false" toShortenFQNames="true">
+  <variable name="LINE_COMMENT_START" expression="lineCommentStart()" defaultValue="" alwaysStopAt="true" />
+  <variable name="DATE" expression="date(&quot;YYYY-MM-dd&quot;)" defaultValue="" alwaysStopAt="true" />
+  <variable name="USER" expression="user()" defaultValue="" alwaysStopAt="true" />
+  <context>
+    <option name="JAVA_CODE" value="true" />
+    <option name="KOTLIN" value="true" />
```
Comment (alexander-yevsyukov) [inline .idea/live-templates/todo.xml:23]: We need JavaScript and Dart too.

Comment (alexander-yevsyukov) [reply .idea/live-templates/todo.xml:23]: Sometimes it is needed in HTML.

Comment (alexander-yevsyukov) [reply .idea/live-templates/todo.xml:23]: And Markdown.

File: quality/detekt-config.yml [hunk 2]
```diff
@@ -20,3 +20,5 @@ style:
   MaxLineLength:
     maxLineLength: 120
```
Comment (alexander-yevsyukov) [inline quality/detekt-config.yml:21]: I think, we have 100 in our code. Don't we?

**Review by alexander-yevsyukov — COMMENTED** [review]
Please see my comments and questions.

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov Please take another look. 

I've dropped context restrictions at all. IDEA will try to find the appropriate `commendStart` and `commentEnd` symbols on its own. It works correctly for all mentioned languages, besides Markdown. It officially doesn't have comments at all, and they are simulated with a [hack](https://www.jetbrains.com/help/idea/markdown.html#comments). In our case, IDEA returns nothing for `commentEnd()` function in Markdown documents, producing a comment without a closing parenthesis. It's not a big penalty taking into account, we would have a single template.

**Comment by yevhenii-nadtochii** [comment]
And, as far as I understand, @armiol suggested putting this `todo.xml` along with `spine.xml` (on Google Drive) and update a dedicated section in Wiki of `documentation` repo?

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Comment by alexander-yevsyukov** [comment]
> And, as far as I understand, @armiol suggested putting this `todo.xml` along with `spine.xml` (on Google Drive) and update a dedicated section in Wiki of `documentation` repo?

As I understood, Alex gave links to the current documentation on how to arrange live templates in IDEA.

**Review by armiol — CHANGES_REQUESTED** [review]
Please see my comment above.

**Comment by yevhenii-nadtochii** [comment]
@armiol @alexander-yevsyukov PTAL. 

I've included `Spine.xml` template group and updated the [corresponding section](https://github.com/SpineEventEngine/documentation/wiki/IntelliJ-IDEA-configuration#live-templates) in wiki. 

Although, I can't delete content from [Spine > spine.io > wiki-resources](https://drive.google.com/drive/u/0/folders/0B562RqcmJm7OZzdOeVVfZVNvNDQ?resourcekey=0-cbM2y3JqU4Gy--PlmNpdIA) Google Drive folder. I would like to delete that two files, as they are not needed anymore.

**Review by armiol — APPROVED** [review]

**Pull request merged**